src/Matcher/Pattern/TypePattern.php

```diff
@@ -82,7 +82,7 @@ public function getError()
     public function hasExpander(string $expanderName)
     {
         foreach ($this->expanders as $expander) {
-            if ($expander->getName() === $expanderName) {
+            if ($expander::NAME === $expanderName) {
```

Comment:
that's the problem, interface can't guarantee that Expander will have `NAME` constant. Please add `is($name)` method as I suggested in my previous comment